features/layer_lock.c

```diff
@@ -21,6 +21,37 @@
 // The current lock state. The kth bit is on if layer k is locked.
 static layer_state_t locked_layers = 0;
 
+#if LAYER_LOCK_IDLE_TIMEOUT > 0
+#if LAYER_LOCK_IDLE_TIMEOUT < 100 || LAYER_LOCK_IDLE_TIMEOUT > 30000
+//Constrain timeout to a sensible range. With the 16-bit timer, the longest
```

Comment:
Style nit: My (implicit, unstated) preference is that comments have a space after the `//`, like

```c
// Constrain timeout to a sensible range...
```